homeassistant/components/tts/__init__.py

```diff
@@ -1038,7 +1038,7 @@ async def post(self, request: web.Request) -> web.Response:
             _LOGGER.error("Error on init tts: %s", err)
             return self.json({"error": err}, HTTPStatus.BAD_REQUEST)
 
-        base = get_url(self.tts.hass)
+        base = get_url(self.tts.hass, prefer_external=True)
```

Comment:
We always prefer local (part of our slogan is local control and privacy first). Therefore, this change doesn't make sense to me, to be honest.

The PR doesn't describe the exact issue it is solving or how this might impact anything else.

Could you provide the use cases in the PR description this fixes (and what it could potentially break)?

../Frenck

Reply:
It says e.g. here: https://www.home-assistant.io/integrations/tts/ 
![image](https://github.com/user-attachments/assets/8f4cc6e3-526d-4081-9316-0f6b7e5540fc)

The probelm is right now, even though setting an external address..it doesn't use it. So the documentation is "broken" and also the tty for google cast in case the internal url is not directly accessable (like it was in my case).

This is kind of a documentation bug in my point of view, but thinking about this longer and also taking your "local first" approach into consideration, I would say a better approach to fix this is:
* Fix the documentation, ..setting an external url has no impact.
* (Re)-Introduce either "base url" or a setting like "prefer external url" which then gets passed to the get_url parameter inside the "tts" config.

I couldn't find the reason why "base url" has been removed from the configs for tts, I guess it was "too complicated"? Would be introducing a boolean flag for prefering the external url over the internal a valid approach for adding to the tts config parameters?

Reply:
> The PR doesn't describe the exact issue it is solving or how this might impact anything else.

Users using HTTPS in LAN, provided through local DNS server, are unable to cast TTS messages to Google devices.

Possible workarounds are either "don't use HTTPS in LAN" or "buy new router and block traffic to 8.8.8.8 from Google devices".